### PR TITLE
WIP: Update the RELEASE doc with notes from last release

### PR DESCRIPTION
**Pull request by cnunciato**
Signed-off-by: Christian Nunciato <chris@nunciato.org>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: RELEASE.md [hunk 1]
```diff
@@ -73,24 +73,18 @@ Highlights:
     $ git checkout -b <branch>
     ```
 
-1. Remove the `-dev` suffix from the version number found in the `VERSION` file. *Note*: there must not be a space after the `-i`.
-
-    ```
-    $ sed -i'' -e 's/-dev//' VERSION
```
Comment (cnunciato) [inline RELEASE.md]: When I ran this on my Mac, I ended up with a file called `VERSION-e` containing `0.62.0-dev`. I think it's clear enough that we say "remove the -dev" and leave it at that. 😄

Comment (christophermaier) [reply RELEASE.md]: I think the `sed` that comes with macOS responds to different options than the GNU one on Linux.

That's fair; pretty soon Expeditor will be doing this for us, anyway. 😄

File: RELEASE.md [hunk 2]
```diff
@@ -40,13 +40,13 @@ classes: body-article
...
 
-We are happy to announce the release of Habitat <VERSION>. We have a number of new features as well as bug fixes, so please read on for all the details. If you just want the binaries, head on over to [Install Habitat](https://www.habitat.sh/docs/using-habitat/#install-habitat). Thanks again for using Habitat!
+We are happy to announce the release of Habitat <VERSION>. We have a number of new features as well as bug fixes, so please read on for all the details. If you just want the binaries, head on over to [Install Habitat](https://www.habitat.sh/docs/install-habitat/). Thanks again for using Habitat!
 
 Highlights:
 * [BREAKING CHANGE] SOME BREAKING CHANGE
 ```
 
-**Important:** Filter all closed PRs labeled `C-breaking` and ensure any and all of these PRs is mentioned as a `[BREAKING CHANGE]` in the post highlights.
+**Important:** Filter all closed PRs labeled `T-breaking` and ensure any and all of these PRs is mentioned as a `[BREAKING CHANGE]` in the post highlights.
```
Comment (christophermaier) [inline RELEASE.md]: Same as above.

File: RELEASE.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ the `hab-launch` binary has a separate release process. See
 
 Start with the [autogenerated CHANGELOG](https://github.com/habitat-sh/habitat/blob/master/CHANGELOG.md) - Copy and paste this into a google doc for collaboration.
 
-Look through the merged pull requests and see if any of them is marked with the [C-breaking] label - if they are, make sure to mark their entry in the release notes with [BREAKING CHANGE].
+Look through the merged pull requests and see if any of them is marked with the [T-breaking] label - if they are, make sure to mark their entry in the release notes with [BREAKING CHANGE].
```
Comment (christophermaier) [inline RELEASE.md]: Looks like this is old process; now, when we add`X-change` to any PR, Expeditor will automatically add an entry to the `Behavioral Changes` section of the changelog. You can just look at that section without having to do any kind of manual filtering at all. Which is cool.

I think we need to revisit some of the labels we have; some seem superfluous now.

File: RELEASE.md [hunk 4]
```diff
@@ -73,24 +73,18 @@ Highlights:
...
-
-1. Commit `VERSION` changes and push your branch
-1. Issue a new PR with the `Expeditor: Exclude from Changelog` label and await approval (in the form of a [dank gif](http://imgur.com/X0sNq)) from two maintainers
-1. Pull master once again once the PR is merged into master
-1. Create & push a Git tag
+1. Remove the `-dev` suffix from the version number found in the `VERSION` file and save.
+1. Commit the `VERSION` change and push your branch.
+1. Issue a new PR with the `Expeditor: Exclude from Changelog` label and await approval (in the form of a [dank gif](http://imgur.com/X0sNq)) from two maintainers.
+1. Pull master once again once the PR is merged into master.
+1. Create & push a Git tag:
```
Comment (christophermaier) [inline RELEASE.md]: Addition of missing periods and colons:
tenor-5456706

**Comment by cnunciato** [comment]
Closing this as super-outdated.

(Also,  👋!)

**Pull request closed without merging**